app/hotels/src/map/hotelDetailPreview/HotelDetailPreview.js

```diff
@@ -0,0 +1,125 @@
+// @flow strict
+
+import * as React from 'react';
+import { View } from 'react-native';
+import {
+  StyleSheet,
+  NetworkImage,
+  Price,
+  Text,
+  Stars,
+} from '@kiwicom/mobile-shared';
+import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
+import { Translation } from '@kiwicom/mobile-localization';
+import idx from 'idx';
+import { defaultTokens } from '@kiwicom/mobile-orbit';
+
+import type { HotelDetailPreview_availability } from './__generated__/HotelDetailPreview_availability.graphql';
+import { HotelDetailConsumer } from './HotelDetailPreviewContext';
+import HotelReviewScore from '../../components/HotelReviewScore';
+
+type Props = {|
+  +availability: ?HotelDetailPreview_availability,
+|};
+
+export const HotelDetailPreview = ({ availability }: Props) => {
+  const name = idx(availability, _ => _.hotel.name);
+  const price = idx(availability, _ => _.price) || {};
+  const image = idx(availability, _ => _.hotel.mainPhoto.thumbnailUrl);
+
+  return (
+    <HotelDetailConsumer>
+      {({ containerWidth }) => (
+        <View style={[styles.container, { width: containerWidth }]}>
+          <View>
+            <NetworkImage
+              style={styles.image}
+              source={{
+                uri: image,
+              }}
+              resizeMode="contain"
+            />
+          </View>
+          <View style={styles.description}>
+            <Text style={styles.hotelName} numberOfLines={1}>
+              <Translation passThrough={name} />
+            </Text>
+            <View style={styles.row}>
+              <Stars
+                rating={idx(availability, _ => _.hotel.rating.stars)}
+                style={styles.stars}
+              />
+              <HotelReviewScore hotel={idx(availability, _ => _.hotel)} />
+            </View>
+            {price &&
+              price.currency != null &&
```

Comment:
I think we would need to refactor this component to accept a price object with properties currency amount and it would handle this kind of logic. I'll create an issue for it.